animations/tween-animations.md

```diff
@@ -0,0 +1,727 @@
+---
+layout: page
+title: Tween Animations in Flutter
+permalink: /animations/tween-animations.html
+---
+
+* TOC Placeholder
+{:toc}
+
+The animation system in Flutter is based on typed
+[`Animation`](https://docs.flutter.io/flutter/animation/Animation-class.html)
+objects. Widgets can either incorporate these animations in their build
+functions directly by reading their current value and listening to their
+state changes or they can use the animations as the basis of more elaborate
+animations that they pass along to other widgets.
+
+## Animation<double>
+
+In Flutter, an Animation object knows nothing about what is onscreen.
+An Animation is an abstract class that understands its current value
+and its state (completed or dismissed). One of the more commonly used
+animation types is Animation<double>.
+
+A few facts about Flutter animations:
+
+* An Animation object in Flutter is a class that sequentially generates
+  interpolated numbers between two values over a certain duration.
+  The output of an Animation object may be linear, a curve, a step function,
+  or any other mapping you can devise. Depending on how the Animation object
+  is controlled, it could run in reverse, or even switch directions in the
+  middle.
+
+* Animations can also interpolate types other than double, such as
+  Animation<Color> or Animation<Size>.
+  You'll learn about other types later in this page.
+
+* An `Animation` object has state. Its current value is always available
+  in the `.value` member.
+
+* An Animation object knows nothing about rendering or `build()` functions.
+  You'll see later how to make the connection.
+
+## AnimationController
+
+AnimationController is a special Animation object that generates a new
+value whenever the hardware is ready for a new frame. By default,
+an AnimationController linearly produces the numbers from 0.0 to 1.0
+during a given duration. For example, this code creates an Animation object,
+but does not start it running:
+
+<!-- skip -->
+{% prettify dart %}
+final AnimationController controller = new AnimationController(
+    duration: const Duration(milliseconds: 2000), vsync: this);
+{% endprettify %}
+
+AnimationController derives from Animation<double>, so it can be used
+wherever an Animation object is needed. However, the AnimationController
+has additional methods to control the animation. For example, you start
+an animation with the `.forward()` method. The generation of numbers is
+tied to the screen refresh, so typically 60 numbers are generated per
+second. After each number is generated, each Animation object calls the
+attached Listener objects.
+
+When creating an AnimationController, you pass it a `vsync` argument.
+This option keeps the current widget tree in display memory until
+Flutter's rendering engine completes its refresh cycle. To use your
```

Comment:
The sentence "This option keeps the current widget tree in display memory until Flutter's rendering engine completes its refresh cycle." is not at all correct. :-)

What vsync does is it ties the AnimationController to a particular place in the widget tree, so that when that part of the widget tree isn't on the screen, the AnimationController stops calling its callback (though without pausing the clock, so time still passes). Specifically, the vsync argument specifies a TickerProvider.

Reply:
What is that "particular place" in the widget tree? Is it the State object from which the TickerProvider is obtained?

Reply:
It ties it to the TickerProvider; typically that's the State that mixes in the TickerProviderStateMixin. TickerProviderStateMixin further defers to the subtree that's rooted at the nearest TickerMode widget.